### DISCUSSION TITLE
Data insert batch size for 512 dim vectors

### DISCUSSION THREAD

**Discussion by Mahanchello** [milvus-io/milvus#21801] [Q&A and General discussion] [answered]
Hi! I have the following situation: 

When i'm trying in insert data into milvus in batches having 512 dim vectors, batch size usually is  around 3000 elements to be inserted, otherwise i get an error `RPC error: [batch_insert], <MilvusException: (code=1, message=Broker: Message size too large)>` when i'm trying to increase it. 
This is a bit slow and i'm wondering if there is way to configure milvus, so i could increase batch size for the data to be inserted? 

Note: i'm using milvus  v2.2, pymilvus v2.2.1

Thank you!

**Comment by yhmo** [comment, marked as answer]
Looks like "Broker: Message size too large" is thrown from Kafka:
https://stackoverflow.com/questions/59322133/kafka-broker-message-size-too-large

When Milvus proxy node receives insert request, it splits the data into pieces by shard number, then passes the data to Pulsar or Kafka. If you are using Kafaka as message storage, you can change the configuration **message.max.bytes** and **max.request.size** for large message size.

**Reply by xiaofan-luan** [reply]
3000 is not a large number, maybe we need to tune the kafka config a little bit?
By cauculation 3000 is the batch number then the log entry would be somewhere around 1 MB
